src/resilience/Config.hpp

```diff
@@ -82,7 +82,12 @@ namespace KokkosResilience
       template< typename T, typename = std::enable_if_t< std::is_constructible< variant_type, T >::value > >
       explicit Value( T &&val )
       {
-        m_variant = std::forward< T >( val );
+        if constexpr ( std::is_constructible_v< std::string, std::add_rvalue_reference_t< T > > )
```

Comment:
I had to driveby fix a bug where `const char *` ptrs where being auto converted to bool with gcc 11 (this is not the case for gcc 12)